bats/sql-reset.bats

```diff
@@ -172,3 +172,80 @@ teardown() {
      [[ "$output" =~ pk ]] || false
      [[ "$output" != 1  ]] || false
 }
+
+@test "DOLT_RESET --hard properly wipes diff system table immediately" {
+      run dolt sql << SQL
+INSERT INTO test VALUES (1);
+SELECT DOLT_RESET('--hard');
+SELECT count(*) FROM dolt_diff_test;
+SQL
+    [ $status -eq 0 ]
+    # Represents that the diff table marks a change from the recent commit.
+    [[ "$output" =~ "0" ]] || false
```

Comment:
What does DOLT_RESET return? Could that select match 0?

Reply:
Easy way to make sure this isn't happening is just to run the count in a separate dolt select invocation